crates/swc_ecma_transforms_compat/src/bugfixes/safari_id_destructuring_collision_in_function_expression.rs

```diff
@@ -0,0 +1,115 @@
+use swc_atoms::JsWord;
+use swc_common::{
+    collections::{AHashMap, AHashSet},
+    Span,
+};
+use swc_ecma_ast::*;
+use swc_ecma_transforms_base::hygiene::rename;
+use swc_ecma_visit::{as_folder, noop_visit_mut_type, Fold, VisitMut, VisitMutWith};
+
+pub fn safari_id_destructuring_collision_in_function_expression() -> impl Fold + VisitMut {
+    as_folder(SafariIdDestructuringCollisionInFunctionExpression::default())
+}
+
+#[derive(Default, Clone)]
+struct SafariIdDestructuringCollisionInFunctionExpression {
+    fn_expr_name: JsWord,
+    destructured_id_span: Option<Span>,
+    other_ident_syms: AHashSet<JsWord>,
+    in_body: bool,
+}
+
+impl VisitMut for SafariIdDestructuringCollisionInFunctionExpression {
+    noop_visit_mut_type!();
+
+    fn visit_mut_binding_ident(&mut self, binding_ident: &mut BindingIdent) {
```

Comment:
This is not var-decl only. The pattern of assignments and the name of variable declarations share the `Pat` type, and binding identifier is used to differentiate it with Identifier reference.